diff --git a/node_modules/@envelop/sentry/esm/index.js b/node_modules/@envelop/sentry/esm/index.js
index 6ab332f..f348887 100644
--- a/node_modules/@envelop/sentry/esm/index.js
+++ b/node_modules/@envelop/sentry/esm/index.js
@@ -22,6 +22,10 @@ export const useSentry = (options = {}) => {
     const skipOperation = pick('skip', () => false);
     const skipError = pick('skipError', defaultSkipError);
     function addEventId(err, eventId) {
+        if(!eventId){
+            return err;
+        }
+
         if (options.eventIdKey === null) {
             return err;
         }
@@ -175,7 +179,7 @@ export const useSentry = (options = {}) => {
                                     const errorPathWithIndex = (err.path ?? [])
                                         .map((v) => (typeof v === 'number' ? '$index' : v))
                                         .join(' > ');
-                                    const eventId = Sentry.captureException(err, {
+                                    const eventId = err.originalError && skipError(err.originalError) ? null : Sentry.captureException(err, {
                                         fingerprint: ['graphql', errorPathWithIndex, opName, operationType],
                                         contexts: {
                                             GraphQL: {
