diff --git a/index.mjs b/index.mjs
index 8ea3275e0dd5d8534d0581453cc7a60356bc0ba8..076cd972f0fe343f85667b9f5d1d282c5551ce46 100644
--- a/index.mjs
+++ b/index.mjs
@@ -2,7 +2,7 @@ import { parse, Kind, TypeInfo, visit, visitWithTypeInfo, GraphQLError, getNamed
 import { __awaiter } from 'tslib';
 import inspect from 'object-inspect';
 import { DepGraph } from 'dependency-graph';
-import { Parser } from 'graphql/language/parser';
+import { Parser } from 'graphql/language/parser.js';
 
 function readDocument(source) {
     const result = {