diff --git a/node_modules/@sentry/hub/esm/scope.js b/node_modules/@sentry/hub/esm/scope.js
index 03e6711..b786671 100644
--- a/node_modules/@sentry/hub/esm/scope.js
+++ b/node_modules/@sentry/hub/esm/scope.js
@@ -365,9 +365,14 @@ var Scope = /** @class */ (function () {
      */
     Scope.prototype._notifyEventProcessors = function (processors, event, hint, index) {
         var _this = this;
-        if (index === void 0) { index = 0; }
+        console.log('notifyEventProcessors index', index);
+        if (index === void 0) {
+            index = 0;
+            console.log('notifyEventProcessors index set to', 0);
+        }
         return new SyncPromise(function (resolve, reject) {
             var processor = processors[index];
+            console.log('processor', processor);
             if (event === null || typeof processor !== 'function') {
                 resolve(event);
             }
