diff --git a/dist/index.js b/dist/index.js
index 4d1647d8b9d037c9e091e8378d042da4cc569d84..4e45feaeff93ca320e388254012427d69f42968e 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -2708,7 +2708,7 @@ function defineConfig({
           // other pages
           asPath
         )}`,
-        image = `https://og-image.the-guild.dev/?product=${websiteName}&title=${encodeURI(
+        image = `https://og-image.the-guild.dev/?product=${websiteName.toUpperCase()}&title=${encodeURI(
           pageTitle
         )}`
       } = frontMatter;
