diff --git a/node_modules/@whatwg-node/fetch/dist/create-node-ponyfill.js b/node_modules/@whatwg-node/fetch/dist/create-node-ponyfill.js
index 9d3b47f..a5116a2 100644
--- a/node_modules/@whatwg-node/fetch/dist/create-node-ponyfill.js
+++ b/node_modules/@whatwg-node/fetch/dist/create-node-ponyfill.js
@@ -142,12 +142,22 @@ module.exports = function createNodePonyfill(opts = {}) {
       const fetch = function (requestOrUrl, options) {
         if (typeof requestOrUrl === "string") {
           // We cannot use our ctor because it leaks on Node 18's global fetch
-          return originalFetch(requestOrUrl, options);
+          return originalFetch(requestOrUrl, options).catch(e => {
+            if (e.cause) {
+              throw e.cause;
+            }
+            throw e;
+          });
         }
         if (requestOrUrl.url.startsWith('file:')) {
           return handleFileRequest(requestOrUrl.url, ponyfills.Response);
         }
-        return originalFetch(requestOrUrl);
+        return originalFetch(requestOrUrl).catch(e => {
+            if (e.cause) {
+              throw e.cause;
+            }
+            throw e;
+          });
       };
 
       ponyfills.fetch = fetch;
