diff --git a/dist/esm/commands/script-loader.js b/dist/esm/commands/script-loader.js
index bcbe1b33d8c4adf3113e561584dbd17c48a197d2..671d1316123c29ea8bc0901e4e8236fbd6290dcd 100644
--- a/dist/esm/commands/script-loader.js
+++ b/dist/esm/commands/script-loader.js
@@ -2,6 +2,7 @@ import { createHash } from 'crypto';
 import { glob, hasMagic } from 'glob';
 import * as path from 'path';
 import * as fs from 'fs';
+import { fileURLToPath } from 'url';
 import { promisify } from 'util';
 const readFile = promisify(fs.readFile);
 const readdir = promisify(fs.readdir);
@@ -349,15 +350,7 @@ async function getFilenamesByPattern(pattern) {
 // Determine the project root
 // https://stackoverflow.com/a/18721515
 function getPkgJsonDir() {
-    for (const modPath of module.paths || []) {
-        try {
-            const prospectivePkgJsonDir = path.dirname(modPath);
-            fs.accessSync(modPath, fs.constants.F_OK);
-            return prospectivePkgJsonDir;
-            // eslint-disable-next-line no-empty
-        }
-        catch (e) { }
-    }
+    return path.dirname(fileURLToPath(import.meta.url));
 }
 // https://stackoverflow.com/a/66842927
 // some dark magic here :-)