diff --git a/dist/index.d.mts b/dist/index.d.mts
index 7cddbae8a44f81705dd25402511e8107ed067650..4728941792e9c07afb5c3d6d80ef3f7c2e3c8157 100644
--- a/dist/index.d.mts
+++ b/dist/index.d.mts
@@ -1271,4 +1271,7 @@ declare function ThemeSwitch({ lite, className }: ThemeSwitchProps): ReactElemen
 
 declare function Layout({ children, ...context }: NextraThemeLayoutProps): ReactElement;
 
-export { Bleed, Collapse, PartialDocsThemeConfig as DocsThemeConfig, Link, LocaleSwitch, Navbar, NotFoundPage, ServerSideErrorPage, SkipNavContent, SkipNavLink, ThemeSwitch, Layout as default, useConfig };
+// KAMIL: needed to enable search in Hive
+declare function Flexsearch({className}: { className?: string }): ReactElement;
+
+export { Flexsearch, Bleed, Collapse, PartialDocsThemeConfig as DocsThemeConfig, Link, LocaleSwitch, Navbar, NotFoundPage, ServerSideErrorPage, SkipNavContent, SkipNavLink, ThemeSwitch, Layout as default, useConfig };
diff --git a/dist/index.js b/dist/index.js
index d72b8265f2ff61699af6135f5861270695ba6e77..900a6a033fb5aedf6b603fe9fd8766350594c7e8 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -3052,5 +3052,7 @@ export {
   Layout as default,
   useConfig,
   useMDXComponents,
-  useTheme3 as useTheme
+  useTheme3 as useTheme,
+  // KAMIL: needed to enable search in Hive
+  Flexsearch
 };
