diff --git a/dist/index.js b/dist/index.js
index 788fff4cb15ddd9ea3d19493cc80dd69951caca6..5efca5acdbf794d5e01d1355d7af37103e46a1c2 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1500,8 +1500,8 @@ var classes3 = {
 function FolderImpl({ item, anchors }) {
   const routeOriginal = useFSRoute3();
   const [route] = routeOriginal.split("#");
-  const active = [route, route + "/"].includes(item.route + "/");
-  const activeRouteInside = active || route.startsWith(item.route + "/");
+  const active = item.route && [route, route + "/"].includes(item.route + "/");
+  const activeRouteInside = active || (route !== '/' && item.route !== '') && route.startsWith(item.route + "/");
   const focusedRoute = useContext5(FocusedItemContext);
   const focusedRouteInside = !!(focusedRoute == null ? void 0 : focusedRoute.startsWith(item.route + "/"));
   const level = useContext5(FolderLevelContext);
