diff --git a/dist/file-system.mjs b/dist/file-system.mjs
index 30a12c4c6dd3f0004f639ba34b07fc2b0d89e4c5..9324e496b0e6488ab79fa731715c80700183496e 100644
--- a/dist/file-system.mjs
+++ b/dist/file-system.mjs
@@ -2,7 +2,9 @@
 import * as findPagesDirImport from "next/dist/lib/find-pages-dir.js";
 import { CWD } from "./constants.mjs";
 import { getDefault } from "./utils.mjs";
-var { findPagesDir, existsSync } = getDefault(findPagesDirImport);
+// KAMIL: next v13.5 does not export existsSync (under the hood it used to use "fs" anyway)
+import { existsSync } from 'fs'
+var { findPagesDir } = getDefault(findPagesDirImport);
 function findPagesDirectory() {
   const res = findPagesDir(CWD, false);
   return res.pagesDir || // next v13
