diff --git a/lib/tarballs/build.js b/lib/tarballs/build.js
index ffbbd21cd58e776f16857a1b0de37366763a0089..be0933cf3614e8a0de7b1c3275e11419d068af8b 100644
--- a/lib/tarballs/build.js
+++ b/lib/tarballs/build.js
@@ -55,11 +55,6 @@ async function build(c, options = {}) {
             await qq.x('yarn --no-progress --production --non-interactive');
         }
         else {
-            let lockpath = qq.join(c.root, 'package-lock.json');
-            if (!await qq.exists(lockpath)) {
-                lockpath = qq.join(c.root, 'npm-shrinkwrap.json');
-            }
-            await qq.cp(lockpath, '.');
             await qq.x('npm install --production');
         }
     };