diff --git a/dist/src/routines/executeQuery.js b/dist/src/routines/executeQuery.js
index 007d169..61441f8 100644
--- a/dist/src/routines/executeQuery.js
+++ b/dist/src/routines/executeQuery.js
@@ -4,7 +4,6 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
 Object.defineProperty(exports, "__esModule", { value: true });
 exports.executeQuery = void 0;
-const get_stack_trace_1 = require("get-stack-trace");
 const p_defer_1 = __importDefault(require("p-defer"));
 const serialize_error_1 = require("serialize-error");
 const constants_1 = require("../constants");
@@ -80,17 +79,6 @@ const executeQuery = async (connectionLogger, connection, clientConfiguration, s
     }
     const queryInputTime = process.hrtime.bigint();
     let stackTrace = null;
-    if (clientConfiguration.captureStackTrace) {
-        const callSites = await (0, get_stack_trace_1.getStackTrace)();
-        stackTrace = callSites.map((callSite) => {
-            return {
-                columnNumber: callSite.columnNumber,
-                fileName: callSite.fileName,
-                functionName: callSite.functionName,
-                lineNumber: callSite.lineNumber,
-            };
-        });
-    }
     const queryId = inheritedQueryId !== null && inheritedQueryId !== void 0 ? inheritedQueryId : (0, utilities_1.createQueryId)();
     const log = connectionLogger.child({
         queryId,
